### PR TITLE
fix: update Firestore rules

### PR DESCRIPTION
**Pull request by tomarra**
## Description

Adding in a few more checks

* Initials have to be 3 in length and only be capital letters
* Score needs to be within the valid score window
* Character needs to match the defined set

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [x] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Comment by eseidelGoogle** [comment]
I believe @erickzanardo already has commit access and should be able to help this land.  @tvolkert or @Hixie can also get you (and your VGV colleagues) access via https://github.com/flutter/flutter/blob/master/CONTRIBUTING.md.

Thanks for your continued work on this, looks great. :)

**Review by tvolkert — APPROVED** [review]

**Comment by creativecreatorormaybenot** [comment]
I like this PR but I think this is not the best conduct. We should have merged #467 and created a new PR for the score change.

We can merge this one anyway for now because I really do not mind, but it is not how I would do OSS.

**Comment by tomarra** [comment]
> I like this PR but I think this is not the best conduct. We should have merged #467 and created a new PR for the score change.
> 
> We can merge this one anyway for now because I really do not mind, but it is not how I would do OSS.

Sorry about that! Was not the intent. I was working on these in parallel it seems.

File: firestore.rules [hunk 1]
```diff
@@ -9,12 +9,20 @@ service cloud.firestore {
       }
       
       function inCharLimit(initials) {
-        return initials.size() < 4;
+        return initials.size() == 3 && initials.matches('[A-Z][A-Z][A-Z]');
```
Comment (RuiMiguel) [inline firestore.rules:12]:
```suggestion
        return initials.matches('[A-Z]{3}');
```

**Review by RuiMiguel — COMMENTED** [review]
just leave a small comment on the regex, but ok 👍

**Pull request merged**